### ISSUE TITLE
IEEE Ontology Standards Working Group Credential Example

### ISSUE THREAD

**Issue by OR13** [w3c/vc-data-model#963]
Can we develop an example that leverages the IEEE Ontology work?

What claims would be best described by this ontology?

Let's start by getting a JSON example of a credential with some claims that we can look for good definitions for...

For example:

- https://github.com/w3c/vc-data-model/issues/935

```json
{
  "@context": [
    "https://www.w3.org/ns/credentials/v2",
  ],
  "id": "http://example.edu/credentials/1872",
  "type": ["VerifiableCredential", "NewCredentialType"],
  "issuer": { 
    "id": "did:example:123", 
     "type": ["Organization", "OrganizationType"] 
   },
  "issuanceDate": "2010-01-01T19:23:24Z",
  "credentialSubject": {
    "id": "did:example:456", 
    "type": ["Person", "JobType"],
    "claimName": "claimValue"
  }
}
```

**Comment by jimschoening1** [comment]
Thank you @OR13.  Great example.  

Let's start with just "Person" above, which would be replaced with the full line: 

 "type": ["http://www.ontologyrepository.com/CommonCoreOntologies/Person", "JobType"],   

Some benefits:
1. The developer doesn't need to define this term, but rather only find it in Common Core Ontology. 
2. It is well defined.
3. It resolves (try it), so a human or machine in any ecosystem can understand and process it.
4. It is logically consistent with other classes/terms such as organization, Jobtype, found in Common Core Ontology and domain ontology extensions, which is critical when data is combined/integrated for new value. 

Current Limitations:
1. It resolves to a temporary URL server (but the IEEE PURL server is in testing and will be up soon)
2. Common Core Ontology is not yet an approved international standard (but is very mature, so common classes such as 'Person' won't likely change) 
3. Many needed terms won't be found in Common Core Ontology or a domain extension ontology (but the most commonly used terms are, and new ones can be submitted for inclusion to the IEEE Ontology Standards Working Group, which I chair)

**Comment by jimschoening1** [comment]
@OR13   Your thoughts on this, pro or con?    Is this in-scope for #953?  Could we move it back there?

**Comment by OR13** [comment]
@jimschoening1 Its not in scope for the vocab thread imo, because these are terms you want to explicitly define in an interoperable way, here is a JSON-LD version of what you are suggestion.

```json
{
  "@id": "did:example:456",
  "@type": "http://www.ontologyrepository.com/CommonCoreOntologies/Person"
}
```

https://v.jsld.org/M2ZuC5j7ApNV2CKd6nzobQzzvkSuUx9s7PUYfCigJCCPbWPkyLnYSyXsJsqriE19EM1cURjv9vmyDDjD3Z2Ghr2SQGw9A3U27vva9uSWM75XQtD3caKxJbj4kijLbbt94wm9x8krnTcPU6a51Q2EKzY9p6AtTXdxxpjGLQbvv1uedsQ5harpq66Wva8hZmkjgc

In this case, your intention is to communicate that `did:example:456` is of type  `.../CommonCoreOntologies/Person`...

Maybe others would want to use https://schema.org/Person because of Google / Search engines.

The nice part of the VC Data Model is that people can choose for themselves, and machines can process the results with existing and well adopted standards.

**Comment by OR13** [comment]
We can expand the example by including specific claims that are consistent with `Person`... for example, using https://schema.org/taxID ...

```json
{
  "@id": "did:example:456",
  "@type": "http://www.ontologyrepository.com/CommonCoreOntologies/Person",
  "https://schema.org/taxID": 55555555
}
```

https://v.jsld.org/Cww2WQeUftssmeL7PSGGtbn6x1BBRwAqh7CyAEn3hpeEErX5njx3W4jvS1eRGkBCWNCAhAbXSjbK2Tbd7hsHXhNbMPNBdMhd2CFTVuHiEfnDeZwVHFKT6PwNQPEgdaXMjLCjZxwv4t7dUjcRvUyGjFxawCnvRNwPRavKQbSuDeY2tiBTUGAQyDQABjWJf7eUCp72xWzHMKF12xaCNtgvYrVgv4KqXy3SoEbP4AG

**Comment by OR13** [comment]
Can you propose some additional semantic structures from `CommonCoreOntologies` that we can use to design a credential?

**Comment by jimschoening1** [comment]
Let first try to resolve this one item. 

Today, developers choose terms from many models and standards (all siloed), or define their own terms, or use terms they don't define.   Other applications can only process such data if mapped in advance, and such mappings grow at a rate of n squared, which doesn't scale.   

Wouldn't it be better if the world had a standard for these terms, so disparate data sources could map to it for data integration and sharing?

**Comment by jimschoening1** [comment]
@OR13 Orrie, would you care to get on a video call to synch up?

**Comment by OR13** [comment]
@jimschoening1 sure, send me an email orie@transmute.industries.

**Comment by jimschoening1** [comment]
@OR13 Orie, 

	Here are a few ontology classes from Common Core Ontology, which resolve to an interim server.

http://www.ontologyrepository.com/CommonCoreOntologies/designativename
http://www.ontologyrepository.com/CommonCoreOntologies/biologicalsex
http://www.ontologyrepository.com/CommonCoreOntologies/citizen
http://www.ontologyrepository.com/CommonCoreOntologies/bus


The IEEE interim PURL server should be ready for our team's testing any day now, but the final IEEE PURL server (with domain name purl.ieee.org) will take longer. 

If interested, all are welcome to join the IEEE Ontology Standards Working Group Kick-off meeting next Wed.  I'll post the open invitation next, which can be openly shared.

Jim

**Comment by jimschoening1** [comment]
All are invited to this open meeting Wed 9th:

IEEE Ontology Standards Working Group Kick-Off Meeting

Meeting Link: https://ieeesa.webex.com/ieeesa/j.php?MTID=m0a1580902fd731e8a143c652bdf97a15

Wednesday, Nov 9, 2022 1:00 pm | 1 hour | (UTC-04:00) Eastern Time (US & Canada)
Meeting number: 2338 835 9801
Password: kUQ2J8Dquf6
Agenda: 1.	Call to Order
2.	Introductions and Affiliation Declarations
a.	Establish Voting Membership
b.	Establish Quorum
3.	Approval of Agenda
4.	Review Working Group P&Ps
a.	Call or Appointment of Officers
5.	IEEE Policies Review
a.	IEEE Call for Patents
b.	IEEE SA Copyright Policy
c.	IEEE SA Participation
6.	Contributions, Presentations, and Discussions
a.	Common Core Ontology v0.31 (Ron Rudnicki, CUBRC)
b.	Cyber Ontology (Casey Rock, US Army)
c.	Opaque Identifiers – (Multiple)
d.	MyOntology opensource project and proposed Study Group (Jim Schoening)
e.     IEEE	PURL Server Jim Schoening
7.	New Business
8.	Adjourn

Join by video system
Dial 23388359801@ieeesa.webex.com
You can also dial 173.243.2.68 and enter your meeting number.

Join by phone
+1-646-992-2010 United States Toll (New York City)
+1-213-306-3065 United States Toll (Los Angeles)

Access code: 233 883 59801

**Comment by jimschoening1** [comment]
@OR13 Casey Rock (who took over the US Army Ontology Program when I retired) and I agree your code can point to anything we can post.   The IEEE interim PURL server should be ready for testing any day now, so we will set up things up there so it works for this type of use.   Also, the ontologies will probably be switching to opaque identities (but where the ontology tools provide the labels in various languages for human understandability), so we'll need to test how that works.  We should be back in a couple weeks with an operational demo.  Let us know if you have any questions.

**Comment by TallTed** [comment]
Boy-oh-boy (no pun intended), is `biologicalsex` fraught with peril.  How are `F` and whatever alternative values exist defined?  I'm pretty sure they planned to handle entities assigned "Female" (XX) or "Male" (XY) at birth (though I bet they don't actually require that the chromosomes be checked), but what about [XXY, X](https://www.scientificamerican.com/article/q-a-mixed-sex-biology/), and other variants yet to be discovered?  A world of dangers awaits!

**Comment by jimschoening1** [comment]
Ted, 

 

        Yes, this is a sensitive topic, so let’s seek consensus using the standardization process.  Here’s how:  

 

1.	Review how IEEE P3195.1 Common Core Ontology (in draft) defines ‘biological sex’ at www.ontologyrepository.com/CommonCoreOntologies/biologicalsex <http://www.ontologyrepository.com/CommonCoreOntologies/biologicalsex>  
2.	Propose your edits by creating a Gitlab issue at https://opensource.ieee.org/cco/CommonCoreOntologies/-/issues
3.	Join IEEE Ontology Standards Working Group to build consensus for your changes.  All are invited to our meeting tomorrow at 1300 EDT and monthly.  See https://sagroups.ieee.org/oswg/

 

Jim Schoening

Chair, IEEE Ontology Standards Working Group

**Comment by TallTed** [comment]
I might have thought what I wrote above would be sufficient for anyone already involved with the IEEE Onotology work to raise such an issue, but I've now done [that work for you](https://opensource.ieee.org/cco/CommonCoreOntologies/-/issues/156).

**Comment by TallTed** [comment]
@jimschoening1 
> All are invited to our meeting tomorrow at 1300 EDT and monthly

The page you linked does not say anything about future meetings.  Please confirm whether by "1300 EDT and monthly" you mean "1300 ET, the second Wednesday of every month"?

**Comment by jimschoening1** [comment]
Ted,

               Thank you for submitting this.  If you don’t get a response by say Tues, I’ll find out which ontologist authored this and ask for their response.   Thanks for you input.

 

               The next monthly meeting will be the week of 12 Dec.  I’ll let you know when it gets scheduled for.  If your input get’s a thorough vetting on the issue tracker, it could be ready for the Dec agenda. 

 

               Thanks again for this contribution. 

 

Jim Schoening

Chair, IEEE Ontology Standards Working Group.

**Comment by jimschoening1** [comment]
Ted, Did you see the response to your comment (on the definition of Biological Sex at https://opensource.ieee.org/cco/CommonCoreOntologies/-/issues/156 ?  It proposes some changes and seeks your feedback.

**Comment by jimschoening1** [comment]
At an Open Wallet Foundation (OWF) session on Tues, someone asked if the OWF wallet should hold personal data.  Daniel Goldscheider and Tracy Kuhrt said yes.   
I emailed Tracy and we are in discussions on next steps.  I suggested we start an OWF slack channel or thread to collecte requirements and principles, then invite proposed solutions.   
This is relevant to W3C VCs, since they include claim data as a payload, which may need to be understood by a verifier from a different domain. Individuals need a standard data model (or ontology) so external data providing and consuming entities can map and transform their disparate data only once to a personal data standard.

**Comment by TallTed** [comment]
@jimschoening1 —
> Ted, Did you see the response to your comment (on the definition of Biological Sex at https://opensource.ieee.org/cco/CommonCoreOntologies/-/issues/156 ? It proposes some changes and seeks your feedback.

Thank you for the ping.  I received email notification of the comment, but hadn't yet reached it in my overflowing inboxes.  GitHub notifications are higher on my rotating review stack, so it got to me here first.  Feedback posted.  :-)

**Comment by jimschoening1** [comment]
Open Wallet Foundation participants are agreeing an all-purpose wallet should store data, including VC claim data.  We're now discussing if a standard data model is needed for this data.  W3C-VCDM does a good job but stops at claim data, so such a personal data model would pick up there.  See this robust discussion on OWF Discord at https://discordapp.com/channels/1022962884864643214/1044329759347331213

**Comment by TallTed** [comment]
@jimschoening1 — I think most if not all participants in this W3C VCDM work will agree with me when I say that we have made an active decision (and I encourage the OWF to adopt this same decision) *NOT* to create a "standard data model" for *claims* data, because at core, we believe that _anyone should be able to use VCs to say anything about anything_ without any need to convince gatekeepers of such a "standard data model" and/or vocabulary/ies of the need to add their desired (or required!) attribute and/or values.

(The [OWF Discord link](https://discordapp.com/channels/1022962884864643214/1044329759347331213) doesn't take me anywhere useful. After requiring me to click past 2 sales pages, it just says I found myself in a strange place without any text channels. I strongly advise that such barriers to entry be re-examined and eliminated if possible, if "robust participation" by new participants is desired by others, as it seems to be by you.)

**Comment by jimschoening1** [comment]
This would be a voluntary personal data standard, so a VC could refer to its terms if desired, could reference any other vocabulary, or make up its own terms and define them.  Any other reasons this is not a good idea?

**Comment by TallTed** [comment]
Ah, so just another ontology/vocabulary, which might be used anywhere. I am generally of the opinion that it's better to use existing ontologies/vocabularies wholesale than to try to reinvent them or to cherry-pick from them, and to contribute improvements to those existing ontologies/vocabularies when they don't satisfy the immediate need. Of course, when there's a complete gap — no-one's every built an ontology/vocabulary to describe the immediate subject matter — then it does make sense to build a new one from scratch.

**Comment by jimschoening1** [comment]
Will be presenting and demonstrating this work to Open Wallet Foundation (Architecture Task Group) on 23 Jan.  The agenda topic will be something like 'Understandable VC Claim Data via Standard Ontologies.'  We will demo how a VC issuer searches on the IEEE standard ontologies, finds desired term, inserts IRI in VC, which resolves to forthcoming IEEE PURL swerver.  Verifier looks up IRI for standard semantics.  The IEEE ontologies already have the most common terms, and new ones can be added to the draft standards within a few months.

**Comment by jimschoening1** [comment]
My Data Global is now actively working on a personal data model.  18 members are subscribed to their Slack channel at standard-personal-data-model on mydataglobal.slack.com.  All are welcome.

**Comment by Fak3** [comment]
That slack is closed for registration or login.

**Comment by mwherman2000** [comment]
From Jim…
        Try the link at Slack (mydata.org)<https://www.mydata.org/slack/>.  Thanks for your interest.
Jim

**Comment by mwherman2000** [comment]
From @jimschoening1 
> Try the link at Slack (mydata.org)<https://www.mydata.org/slack/>.  Thanks for your interest.
> Jim

**Comment by OR13** [comment]
Semi related: https://github.com/w3c/vc-data-model/issues/1007

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-04-04-vcwg-special#section1-6) on 2023-04-04

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.6. IEEE Ontology Standards Working Group Credential Example (issue vc-data-model#963)

_See github issue [vc-data-model#963](https://github.com/w3c/vc-data-model/issues/963)._




**Kristina Yasuda:** About IEEE Ontology Standards. I think we need Orie to see if he would volunteer..  

</details>

**Comment by OR13** [comment]
I suggest closing this if there is no intention to provide exampled on industry ontologies.

I think the current examples in the spec are weak, but we don't need to add an example based on IEEE unless there is strong consensus to do so.

I am adding the pending close label.

**Comment by brentzundel** [comment]
No objections raised since marked `pending close`, closing.
